GES-PersonalProject/Assets/Scripts/Checkpoint.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Checkpoint : MonoBehaviour
+{
+
+    /*[SerializeField]
+    private Color inactivatedColor, activatedColor;*/
+
+    [SerializeField]
+    private float inactivatedScale = 1, activatedScale = 1.1f;
+
+    private SpriteRenderer spriteRenderer;
+
+    private bool isActivated = false;
+
+    private void Start()
+    {
+        spriteRenderer = GetComponent<SpriteRenderer>();
+        //UpdateColor();
+    }
+
+    /*private void UpdateColor()
+    {
+        Color color = inactivatedColor;
+
+        if (isActivated)
+            color = activatedColor;
+
+        spriteRenderer.color = color;
+    }*/
+
+    private void UpdateScale()
+    {
+        float scale = inactivatedScale;
+        if (isActivated)
+            scale = activatedScale;
+
+        transform.localScale = Vector3.one * scale;
+    }
+
+    public void SetIsActivated(bool value)
+    {
+        isActivated = value;
+        UpdateScale();
+        //UpdateColor();
+    }
+
+    private void OnTriggerEnter2D(Collider2D collision)
+    {
+        if (collision.CompareTag("Player"))
+        {
+            Debug.Log("Player entered the checkpoint.");
+            CharacterController player = collision.GetComponent<CharacterController>();
+            player.SetCurrentCheckpoint(this);
+        }
+    }
+}
```

Comment:
Looks good. If any of that commented out code is not being used anymore, you might as well delete it. If you don't want to do that, copy it to a Word doc.

Reply:
Everything looks understandable by a glance and if you're certain that you won't need the commented code you can delete it